### DISCUSSION TITLE
VerticaPy version 1.0.0

### DISCUSSION THREAD

**Discussion by oualib** [vertica/VerticaPy#904] [Announcements]
This is our very first major release, which includes numerous modifications to the previous code aimed at enhancing the user experience.
  
:warning: Please note that this list may be incomplete, and for a comprehensive overview, including additional features, refer to the changelogs.

## CI/CD

The CI/CD pipeline is now more comprehensive, incorporating the following:

 - Integration of **Black** for code formatting.
 - Inclusion of **Pylint** to assess code robustness with a minimal threshold, ensuring code can be merged.
 - Implementation of **Pytest** for optimized Unit Tests, with results compared to standard libraries such as sklearn or scipy.
 - Integration of **Codecov** to monitor code coverage.
 - Implementation of **GitHub actions** across different Python versions (3.9-3.12) to ensure support for all of them.

## New Features

Some of the highlights of this release are:

**Algorithms**

 - IsolationForest
 - KPrototypes
 - Poisson Regression
 - AutoRegressive (AR)
 - MovingAverages (MA)
 - AutoRegressive Moving Averages (ARMA)
 - AutoRegressive Integrated Moving Averages (ARIMA)
 - Term Frequency * Inverse Document Frequency (TFIDF). It is still beta.

**ML New features**

 - New function for finding the feature importance for XGBoost models.
 - Classification metrics are now available for multiclass data/model using three methods: micro, macro, weighted, score and none. - average_precision_score is another new metric that is added to classification metrics. - roc_auc and prc_auc now work for multi-class classification using different averaging techniques stated above.
 - Model names are now optional
 - Model Tracking and Versioning now supported. Check out notebooks/ml/model_tracking_versioning/index.ipynb for more details.
 - Model Export and Import: Now models can be exported to pmml, tensorflow, and binary. They can now be exported to another User Defined Location.
 - Added Model Versioning / Tracking (MLOps)

**SQL**

 - vDataFramesSQL is deprecated. Now, vDataFrame can be used directly to create vDataFrames from SQL.

**Plotting**

- Flexibility to select from the three renowned plotting libraries for professional plots: Matplotlib, Highcharts, Plotly.

**Library Hierarchy**

 - Import structures have changed. The code has been completely restructured, which means that going forward all imports will be done differently. Currently, we still allow the previous structure of import, but it will gradually be deprecated.

## Website

A new website will be soon available. It will include all the important information about the different changes and a totally new documentation generated by Sphinx.

## Release Notes

**Changelogs**

 - For a comprehensive list of all the changes, please refer to the [change log](https://github.com/vertica/VerticaPy/blob/master/change_logs/change_log_v1.0.0.md).
 - Full Changelog: https://github.com/vertica/VerticaPy/compare/0.12.0...1.0.0

**Installation**

The release will be on available on the defaults and can be installed using:

```
python3 -m pip install verticapy
```

If you want to install extra features, use:

```
python3 -m pip install verticapy[all]
```

Please report any issues on our [GitHub page](https://github.com/vertica/VerticaPy)

**Contributors**

We would like to extend our thanks to all the contributors who made this release possible:

- Badr Ouali @oualib
- Umar Farooq Ghumman @mail4umar
- Arash Fard @afard
- Tyler Consigny @tconsigny
- Vikash Singh @vikash018
- Abhishek Sharma @abhsharma2 
- Matteo Monaldi @mat-shyR 
- Abdelhak Zabour @VerticaDataScientist 

If you would like to contribute then please visit our updated [contributing guidelines](https://github.com/vertica/VerticaPy/blob/master/CONTRIBUTING.md).

<hr /><em>This discussion was created from the release <a href='https://github.com/vertica/VerticaPy/releases/tag/1.0.0'>VerticaPy version 1.0.0</a>.</em>
